plugin-server/src/utils/event.ts

```diff
@@ -1,38 +1,58 @@
 import { PluginEvent, ProcessedPluginEvent } from '@posthog/plugin-scaffold'
 import { KafkaMessage } from 'kafkajs'
 
-import { ClickhouseEventKafka, IngestionEvent } from '../types'
+import { ClonableIngestionEvent, Event, PostIngestionEvent, RawEvent } from '../types'
 import { chainToElements } from './db/elements-chain'
 import { personInitialAndUTMProperties } from './db/utils'
-import { clickHouseTimestampToISO } from './utils'
-
-export function convertToProcessedPluginEvent(event: IngestionEvent): ProcessedPluginEvent {
-    const timestamp = typeof event.timestamp === 'string' ? event.timestamp : event.timestamp.toUTC().toISO()
+import { clickHouseTimestampToDateTime } from './utils'
 
+export function convertToProcessedPluginEvent(
```

Comment:
Untested changes here. This kind of code is easy to test - we should do that by default.